src/WebHandlers.cpp

```diff
@@ -205,9 +205,13 @@ void AsyncStaticWebHandler::handleRequest(AsyncWebServerRequest *request)
       return request->requestAuthentication();
 
   if (request->_tempFile == true) {
-    time_t lw = request->_tempFile.getLastWrite();    // get last file mod time (if supported by FS)
-    if (lw) setLastModified(gmtime(&lw));
-    String etag(lw ? lw : request->_tempFile.size());   // set etag to lastmod timestamp if available, otherwise to size
+    #if defined(ESP32) || defined(ESP8266)
+      time_t lw = request->_tempFile.getLastWrite();    // get last file mod time (if supported by FS)
+      if (lw) setLastModified(gmtime(&lw));
+        String etag(lw ? lw : request->_tempFile.size());   // set etag to lastmod timestamp if available, otherwise to size
+    #else
+      String etag = String(request->_tempFile.size());
```

Comment:
@ayushsharma82 : `getLastWrite()` is not supported on rp2040 ?

Reply:
It was but gave an error related to something ambiguous `String(long long int)`. Can you check why that’s happening? It could be an bug in arduino-pico repo.

Reply:
It's because the current code is wrong: `String etag(lw ? lw : request->_tempFile.size()); ` lw is time_t (which is long on esp32) and size is unsigned int (which can be int).

The code could be fixed as-is:

`String etag = lw ? String(lw) : String(request->_tempFile.size());`

Reply:
Still the same:

```
WebHandlers.cpp:213:35: error: call of overloaded 'String(time_t&)' is ambiguous
  213 |       String etag = lw ? String(lw) : String(request->_tempFile.size());
  ```

Reply:
ok so it helped find the issue. on rp2049, time_t  is then not a long. I's just a matter of casting it to the right type I think because right now there is no String constructor matching the required type.
if i am not mistaken, long long int is int64_t so maybe `String((int64_t) lw)` ?
any way.... if you don't succeed, I will append a commit in the branch to check tomorrow ;-)

Reply:
Still the same. I'll leave it to you for tomorrow, I'm throwing darts in dark :p

Reply:
ok i found a solution